src/test/java/jenkins/plugins/office365connector/TimeUtilsTest.java

```diff
@@ -13,15 +13,19 @@
     public void dateToString_ReturnsDateAsString() {
 
         // given
+        // assuming each element has different value, assertion can validate existence
+        // without worrying about order which depends of locale and timezone
         long date = 1506717341515L;
 
         // when
         String format = TimeUtils.dateToString(date);
 
         // then
         // validates only the date as the timezone and local can be different
-        assertThat(format).startsWith("2017-09-29");
-        // because of the timezone, hours depends whete the test is executed
+        assertThat(format).contains("2017");
+        assertThat(format).contains("09");
+        assertThat(format).contains("29");
```

Comment:
Am not sure if this fixes the issue. Still the expected and the actual output is different.

Reply:
Removed one more assertion as it strongly depends of the locale or timezone. On the other hand I don't want to set timezone or mock it as the output of this method strongly depends just from that variable. So mocking is making useless this test IMO

Reply:
You need to mock the locale and test for different locale the format is correct. Just commenting the day is not enough. If the test is executed on end of month, then even month changes. If executed at the end of the year even the year changes. 
As even doing that would be just like testing the library function, I don't see the point of this test case itself.

Reply:
Good point, current approach is very naive. Will apply suggestion provided by @casz then it should be enough to validate all elements or even whole string at once.